src/sources/apache_metrics/mod.rs

```diff
@@ -34,9 +34,11 @@ use vector_core::config::LogNamespace;
 #[derive(Clone, Debug)]
 pub struct ApacheMetricsConfig {
     /// The list of `mod_status` endpoints to scrape metrics from.
+    #[configurable(metadata(docs::examples = "http://localhost:8080/server-status/?auto"))]
     endpoints: Vec<String>,
 
     /// The interval between scrapes, in seconds.
+    #[configurable(metadata(docs::type_unit = "seconds"))]
     #[serde(default = "default_scrape_interval_secs")]
     scrape_interval_secs: u64,
```

Comment:
This field should be changed to `Duration`. You can see of an example of what this looks like here: https://github.com/vectordotdev/vector/blob/master/src/sinks/prometheus/exporter.rs#L120-L122.

Essentially, we leave the field name as-is and simply instruct `serde` to deserialize it as seconds with the use of some macro annotations, which keeps things as-is from the user perspective, but lets us use the strongly-typed `Duration` without also needing to manually specify type units and so on.

You'll also need to add `#[serde_as]` (and add an import for it) to `ApacheMetricsConfig` itself.

Reply:
Oh that is good to know. I guess we'll be doing that for the `http_client` source as well.

Reply:
There's a particular case where we can't yet do this conversion, which is for optional fields i.e. `something_secs: Option<u64>`. That's an outstanding limitation with the copious amounts of boilerplate glue code between `serde`, `serde-with`, and `Configurable`.

Any fields that are simply `something_<time unit>: <integer>` or whatever... yeah, we should convert them. This applies to fields that take either seconds or milliseconds, and either `u64` or `f64`, as both `DurationSeconds<T>` and `DurationMilliSeconds<T>` exist, and the `T` there can be `u64` or `f64`.